releasenotes/notes/kubernetes_node_annotations_as_hostalias-e350f7b84567cb6d.yaml

```diff
@@ -0,0 +1,12 @@
+# Each section from every release note are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Add ``kubernetes_node_annotations_as_host_aliases`` parameter to specify a list
+    of Kubernetes node annotations that should be used as host aliases.
```

Comment:
I think we should also mention that `cluster.k8s.io/machine` will be considered a host alias by default